src/meta/src/cluster/mod.rs

```diff
@@ -96,12 +96,13 @@ where
     /// (via `activate_worker_node`) to set its state to `Running`.
     pub async fn add_worker_node(
         &self,
-        host_address: HostAddress,
         r#type: WorkerType,
-    ) -> Result<(WorkerNode, bool)> {
+        host_address: HostAddress,
+        worker_node_parallelism: usize,
+    ) -> Result<WorkerNode> {
         let mut core = self.core.write().await;
         match core.get_worker_by_host(host_address.clone()) {
-            Some(worker) => Ok((worker.to_protobuf(), false)),
+            Some(worker) => Ok(worker.to_protobuf()),
```

Comment:
We may return error here.

Reply:
We can discuss about this later.

Reply:
CN will re-register after it restarts. I think we should check and update worker_node_parallelism here if CN already exist. This indicates that the CN's configuration is updated.